### PR TITLE
backend: add API call to export calendar in ICS format

### PR DESCRIPTION
**Pull request by nvgoldin**
The new route is at: /schedule, with query parameter `download`:
1. /schedule/?download=ical
Will download the existing user scheduled courses in ICS format
(iCalendar)

2. /schedule/?download=json
Will download the existing user scheduled courses in JSON format
but not as repeating events (just the canonical dates)

The ICS format can be easily exported to google calendar, outlook, etc.

No FE here (yet) - would be helpful if you could help testing it out, I tested the basic functionally manually (one course, 2 courses, etc) - and imported it to google calendar. I didn't test all the edge cases: end/begin of semester course, etc.

### How to run
When logged in just add in your browser /schedule/?download=ical
Go to google calendar -> press add new calendar -> press import

Closes https://github.com/asaf-kali/coursist/issues/94

### CODE REVIEW

File: academic_helper/logic/schedule_export.py [hunk 1]
```diff
@@ -0,0 +1,114 @@
+from academic_helper.models import ClassSchedule
+import arrow
+from ics import Calendar, Event
+
+
+# TODO: needs to be modeled in the DB
+_HUJI_SEMESTER_DATES = {
+    1: {
+        "begin": arrow.get(year=2019, month=10, day=27),
+        "end": arrow.get(year=2020, month=1, day=28),
```
Comment (asaf-kali) [inline academic_helper/logic/schedule_export.py:10]: Why using arrow? No need for an external library for that, right?

File: academic_helper/views/schedule.py [hunk 2]
```diff
@@ -30,7 +36,25 @@ def get(self, request, *args, **kwargs):
...
+                buffer, as_attachment=True, filename=f"coursist_export_{self.user}_{arrow.utcnow()}.ics",
+            )
+        elif request.GET.get("download") == "json":
+            event_json = ScheduleExport(user=self.user).as_dict()
+            buffer = BytesIO()
+            buffer.write(json.dumps(event_json, indent=4, sort_keys=True, cls=DjangoJSONEncoder).encode())
+            buffer.seek(0)
+            return FileResponse(
+                buffer, as_attachment=True, filename=f"coursist_export_{self.user}_{arrow.utcnow()}.json",
+            )
```
Comment (asaf-kali) [inline academic_helper/views/schedule.py:54]: Put these before the first lines of the function (no need to generate `response` if it not used, right?).

File: academic_helper/views/schedule.py [hunk 3]
```diff
@@ -81,13 +105,13 @@ def post(self, request, *args, **kwargs):
         if "search_val" in request.POST:
             search_val = request.POST["search_val"]
             return self.on_search_course(search_val)
-        if "course_number" in request.POST:
+        elif "course_number" in request.POST:
```
Comment (asaf-kali) [inline academic_helper/views/schedule.py:108]: There is a return before the `if`, so no need for `elif`, though I agree it is more readable.

File: requirements.txt [hunk 4]
```diff
@@ -13,3 +13,5 @@ sentry-sdk==0.14.4
 django-contrib-comments==1.9.2
 django-s3-storage==0.13.2
 watchtower==0.7.3
+arrow==0.14.7
```
Comment (asaf-kali) [inline requirements.txt:16]: make sure arrow is really need

File: academic_helper/views/schedule.py (hunk 3 above)
Comment (nvgoldin) [reply academic_helper/views/schedule.py:108]: Oh yhe, actually just leftovers(I considered adding it as a POST at one point).

File: academic_helper/views/schedule.py (hunk 2 above)
Comment (nvgoldin) [reply academic_helper/views/schedule.py:54]: Good point - will fix.

File: academic_helper/logic/schedule_export.py (hunk 1 above)
Comment (nvgoldin) [reply academic_helper/logic/schedule_export.py:10]: Good point - originally I thought it will have more usage than the standard datetime. Generally `arrow` is much more flexiable and feature-rich than `datetime` IMO.

Anyways - the only thing that avoids removing it is the `.shift(weekday=)` functionality - I don't think `datetime`/`timedelta` provides that out-of-the-box, see line 100 in this file.

I can try working around it - let me know what you prfer.

File: academic_helper/views/schedule.py (hunk 2 above)
Comment (nvgoldin) [reply academic_helper/views/schedule.py]: Done in https://github.com/asaf-kali/coursist/pull/113/commits/47a24458496f700d47ad8dd3bba70fa2b9ef9cd6

File: requirements.txt (hunk 4 above)
Comment (nvgoldin) [reply requirements.txt:16]: lets discuss @ https://github.com/asaf-kali/coursist/pull/113#discussion_r452329604

File: academic_helper/logic/schedule_export.py (hunk 1 above)
Comment (asaf-kali) [reply academic_helper/logic/schedule_export.py:10]: Ok. I didn't really dive into the logic. If you think its better this way, leave it like that.
